x-pack/plugins/infra/public/pages/metrics/hosts/components/tabs/alerts/alerts_tab_content.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import {
+  getAlertSummaryTimeRange,
+  ObservabilityAlertStatusFilter,
+  useTimeBuckets,
+} from '@kbn/observability-plugin/public';
+import { AlertConsumers } from '@kbn/rule-data-utils';
+import { InfraClientCoreStart, InfraClientStartDeps } from '../../../../../../types';
+import { useUnifiedSearchContext } from '../../../hooks/use_unified_search';
+
+import {
+  ALERTS_PER_PAGE,
+  ALERTS_TABLE_ID,
+  casesFeatures,
+  casesOwner,
+  infraAlertFeatureIds,
+} from '../config';
+import { useAlertsQuery } from '../../../hooks/use_alerts_query';
+
+export const AlertsTabContent = React.memo(() => {
+  const { services } = useKibana<InfraClientCoreStart & InfraClientStartDeps>();
+
+  const { alertStatus, setAlertStatus, alertsEsQueryByStatus } = useAlertsQuery();
+
+  const { unifiedSearchDateRange } = useUnifiedSearchContext();
+
+  const timeBuckets = useTimeBuckets();
+
+  const { application, cases, charts, triggersActionsUi } = services;
+
+  const {
+    alertsTableConfigurationRegistry,
+    getAlertsStateTable: AlertsStateTable,
+    getAlertSummaryWidget: AlertSummaryWidget,
+  } = triggersActionsUi;
+
+  const CasesContext = cases.ui.getCasesContext();
+  const uiCapabilities = application?.capabilities;
+  const casesCapabilities = cases.helpers.getUICapabilities(uiCapabilities.observabilityCases);
+
+  const summaryTimeRange = getAlertSummaryTimeRange(unifiedSearchDateRange, timeBuckets);
+
+  const chartThemes = {
+    theme: charts.theme.useChartsTheme(),
+    baseTheme: charts.theme.useChartsBaseTheme(),
+  };
+
+  return (
+    <EuiFlexGroup direction="column" gutterSize="m">
+      <EuiFlexItem>
+        <ObservabilityAlertStatusFilter onChange={setAlertStatus} status={alertStatus} />
```

Comment:
Another issue I found is when flipping between the statuses, the table re-renders completely when clicking "show all" which causes the page to move away from the tab. Not sure if there's an easy way to mitigate this problem by staying on the scroll position, which would mean keeping the container height fixed for the table temporarily as the user moves through the status filters.


https://user-images.githubusercontent.com/4104278/217803422-3e4d94cb-9123-4dd4-8ec4-c6bb98ce59e6.mp4

Reply:
I could reserve some minimum space, but since we don't have control over this component we don't know the exact size of the table when it renders or reload the data.
Also, if we reserve some minimum space and we don't have results, it could appear weird to have a lot of white space.

Reply:
Yeah, it's not easy - we had similar issues in the APM transaction detail page for moving between each trace and we solved it by dynamically keeping the height. Not sure if that's the best direction, but I much prefer unnecessary white-space temporarily than the jumping experience.

![CleanShot 2023-02-09 at 13 11 41](https://user-images.githubusercontent.com/4104278/217809631-d5669af8-73a4-4463-8a1c-faafb8af6bc8.gif)

Reply:
Thanks for the suggestion, I just pushed a change similar to how is done in the APM traces and it now keeps track of the height. The empty space in case of changes is not elegant, but I agree with you the layout shift effect is worse.

Reply:
@tonyghiani Thanks! Works much better when moving between the status filters now. I did notice something upon my first interaction with the tabs. When clicking on the Alerts tab it jumps to the top of the page.


https://user-images.githubusercontent.com/4104278/217826023-da561239-278c-4441-b60a-fc632ac8b07f.mp4

Feel free to open a separate issue if this something larger, don't want to continue to block this, because I see this as a smaller bug.